Fixes for your profdata support

AliSoftware opened the pull request:
This PR:

* Fixes escaping of Xcode path
* Ensures `slather setup` was passed an xcodeproj argument
* adds an `--input-format` option to `slather setup` too so that it injects `CLANG_ENABLE_CODE_COVERAGE` instead of `GCC_GENERATE_TEST_COVERAGE_FILES`
* prepare the code to update the xcschemes in order to auto-activate "Gather coverage data"

---

Note that modifying the xcscheme files will need CocoaPods/Xcodeproj#288 which was included in xcodeproj 0.27 and above, so we'll need to bump the dependency on `xcodeproj` up to `~> 0.27.0` to be able to use this.  
Since bumping the `xcodeproj` dependency will also requires us to bump the dependency on the `cocoapods` gem as well, it seemed like a big change with potential repercussions so I stopped my implementation here (the code using Xcodeproj 0.27 has successfully been tested on a sample project though and works with `xcodeproj-0.27.3`).

So I thought maybe we should keep that version bump for a separate PR for later, so we can have a first version supporting profdata first that only warns the user about having to activate "Gather coverage data" manually in the meantime.

\cc @mattdelves @marklarr

mattdelves commented:
This looks good to me. @AliSoftware have you run the tests on it? I know there were some failing tests when I ran them a while back and haven't gotten around to fixing them. Should get a chance to look at it tonight Australia time.

marklarr commented:
Thanks guys! I'll take a look later tonight too.

AliSoftware commented:
@mattdelves I have to admit I haven't (given there is no `rake` task for them, I wonder if you guys use `rspec` or `bacon` to run your specs, btw?).
It's already 2am here so… I'll let you the honors of running them :smile: :zzz:

marklarr commented:
the commend `rspec` should be good for running tests... I'll run 'em tonight :grinning:

mattdelves commented:
Yeah, from what I know, running ```bundle exec rspec spec``` should get them running. I would like a rake test or rake spec command in there.

AliSoftware commented on lib/slather/project.rb:
```diff
@@ -111,8 +136,9 @@ def profdata_llvm_cov_output
       xcode_path = `xcode-select -p`.strip
-      llvm_cov_command = File.join(xcode_path, "Toolchains/XcodeDefault.xctoolchain/usr/bin/llvm-cov show -instr-profile #{coverage_profdata} #{binary_file}")
-      `#{llvm_cov_command}`
+      llvm_cov_path = File.join(xcode_path, "Toolchains/XcodeDefault.xctoolchain/usr/bin/llvm-cov")
```
Maybe using `xcrun llvm-cov` directly (instead of building `llvm_cov_path` ourselves and have to `shellescape` it) would be even better?

AliSoftware commented on lib/slather/project.rb:
```diff
@@ -2,14 +2,37 @@
-    def slather_setup_for_coverage
+    def slather_setup_for_coverage(input_format = "gcov")
       build_configurations.each do |build_configuration|
         build_configuration.build_settings["GCC_INSTRUMENT_PROGRAM_FLOW_ARCS"] = "YES"
```
Is this build setting still needed when using Xcode 7's new coverage format? I'm not sure we'll have to test and maybe we can get rid of it when using `profdata`

AliSoftware commented:
Just added 3 commits to this PR:

* Added the ability to auto-detect the format (for both `slather setup` and `slather coverage`) based on the Xcode version used
* Don't make `slather setup` set `GCC_INSTRUMENT_PROGRAM_FLOW_ARCS` in the build settings anymore when we use `--format clang`
* Using `xcrun llvm-cov` instead of building its path ourselves

:warning:  There are still 7 tests failing (I also added a `rake spec` task to the `Rakefile` btw). I haven't added specs for my changes either (esp. the "auto" mode, which has only been tested manually against a sample project)

PS: If you're ok we my own work, maybe you could merge my branch in yours so that my commits gets added to the PR venom/slather#99 and we can continue working on it from there, fixing the tests and all ('cause I'm not sure I'm gonna be able to work much more on it in the upcoming days, so better merge in 99 and give the ability to others to push on top of my commits, right?)

mattdelves commented:
LGTM. Thanks @AliSoftware - I'll take a good look at the tests tonight / over the weekend.
